### DISCUSSION TITLE
Extension of JsPsych image-slider-response

### DISCUSSION THREAD

**Discussion by Tanush99** [jspsych/jsPsych#2444] [Q&A]
Hi everyone,

I actually was working on creating an experiment that involves accessing the value of the slider element in the JsPsych image-slider-response plugin, and changing the stimulus in real time as the slider value changes. This is such that as the participants drag the slider, they should be able to see the stimulus changing in real time. I know based on previous discussion threads that it is possible to display the value of the slider as it is being dragged in real time, but is it currently possible to change the stimulus in accordance to dragging the slider element in real time?

Additionally, I was just wondering if it was also possible to change the color of the slider element. Currently, the color seems to be blue as shown here:
![image](https://user-images.githubusercontent.com/63757371/148884783-65a20f40-dc6e-402a-98c1-84907f0f5115.png)

Where exactly is the color parameter located in the JavaScript file defining the image-slider-plugin as I wasn't able to locate it. Is this color parameter defined in some other file?

Thank you very much for your time, and any help would be much appreciated!

**Comment by jodeleeuw** [comment]
> is it currently possible to change the stimulus in accordance to dragging the slider element in real time?

Not with that particular plugin. The [reconstruction plugin](https://www.jspsych.org/7.1/plugins/reconstruction/) does allow the subject to change the stimulus in real time, but it is set up to use the keyboard right now. You would need to modify the plugin to enable slider-based control.

> Where exactly is the color parameter located in the JavaScript file defining the image-slider-plugin as I wasn't able to locate it. Is this color parameter defined in some other file?

This is the browser's default styling for an `<input type="range">` element. If you have the `jspsych.css` file included in your experiment it *should* override this style to look like this:

![image](https://user-images.githubusercontent.com/595524/149185061-2f7182ac-dd7e-4244-b753-5aea14195bf7.png)

You can customize the CSS for the `<input type="range">` element if you'd like to. Here's a [helpful tool](https://www.cssportal.com/style-input-range/).
